apps/angular/crud/src/app/services/todo.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable, inject } from '@angular/core';
+import { BehaviorSubject, Observable, tap } from 'rxjs';
+import { Todo } from '../types/todo';
+import { HttpClient } from '@angular/common/http';
+import { randText } from '@ngneat/falso';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TodoService {
+  private readonly http = inject(HttpClient);
+
+  private todosList = new BehaviorSubject<Todo[]>([]);
+  public todoList$: Observable<Todo[]> = this.todosList.asObservable();
```

Comment:
You can use signals now 😉

And for better code separation, I think it's better to have your http request inside a separate service as your state. 

And setting the state inside the tap operator feel weird. I don't know if that's very maintainable on entreprise application, but that's my personal preference.

Reply:
Would it be ok if i set the state not of the current service, but of different one inside the tap operator?

Or, if the problem is with using tap for state management, what is the better way to do it?

Reply:
the best way is to have a state service that is calling your http service. But the http service is only dealing with the request. I think I prefer to have the subscribe inside the service instead of outside. That's many the thing that bother me.

Reply:
Thank you very much for your answers

Reply:
> the best way is to have a state service that is calling your http service. But the http service is only dealing with the request. I think I prefer to have the subscribe inside the service instead of outside. That's many the thing that bother me.

If we subscribe within the service, then how can we unsubscribe? I looked at other answers and didn't find an example.

Reply:
The service is linked to a component life

Reply:
The service is created as a singleton and lives with the application all the time. Or I'm wrong ? Thanks for answers.

Reply:
If you don't provide it at the component level, you are right. 
You should read my last article
https://medium.com/p/cc210f7040bd